terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/cache.ts

```diff
@@ -278,36 +278,77 @@ export async function redisLocked<T>(
-      return await client.get(queryKey);
-    });
-
-    if (experimentValue !== undefined && experimentValue !== null) {
-      const numValue = Number(experimentValue);
-      if (!isNaN(numValue)) {
-        console.debug(`Found experiment ${queryKey} with value ${numValue}`);
-        return numValue;
+    console.debug(`Checking redis entry for experiment ${experimentKey} (${queryKey})`);
+    try {
+      const experimentValue: string | undefined | null = await redisPool.use(async (client: RedisClientType) => {
+        return await client.get(queryKey);
+      });
+
+      if (experimentValue !== undefined && experimentValue !== null) {
+        return experimentValue;
       } else {
-        console.warn(`Experiment ${queryKey} found but value is not a valid number: ${experimentValue}`);
+        console.debug(`Experiment ${queryKey} not found, using default value ${defaultValue}`);
       }
+    } catch (e) {
+      console.error(`Error retrieving experiment ${queryKey}, using default value ${defaultValue}: ${e}`);
+    }
+
+    return defaultValue;
+  });
+}
+
+export async function getExperimentValue(experimentKey: string, defaultValue: number): Promise<number> {
+  try {
+    const experimentValue = await getExperimentValueStr(experimentKey, defaultValue.toString());
+    const numValue = Number(experimentValue);
+    if (!isNaN(numValue)) {
+      console.debug(`Found experiment ${experimentKey} with value ${numValue}`);
+      return numValue;
     } else {
-      console.debug(`Experiment ${queryKey} not found, using default value ${defaultValue}`);
+      console.warn(
+        `Experiment ${experimentKey} found but value is not a valid number:` +
+          ` ${experimentValue} - returning default value ${defaultValue}`,
+      );
     }
   } catch (e) {
-    console.error(`Error retrieving experiment ${queryKey}: ${e}`);
+    /* istanbul ignore next */
+    console.error(`Error retrieving experiment ${experimentKey} - returning default value ${defaultValue}: ${e}`);
   }
-
   return defaultValue;
 }
 
+export async function getJoinedStressTestExperiment(experimentKey: string, runnerName: string): Promise<boolean> {
+  const runnerNameSuffix = await getExperimentValueStr('RUNNER_NAME_SUFFIX', '');
+  if (runnerNameSuffix === undefined || runnerNameSuffix === null || runnerNameSuffix === '') {
+    console.debug(`Experiment ${experimentKey} check ignored, as RUNNER_NAME_SUFFIX is not set`);
+    return false;
+  }
+
+  if (!runnerName.endsWith(runnerNameSuffix)) {
+    console.debug(
+      `Runner name ${runnerName} does not match suffix ${runnerNameSuffix} when checking experiment ${experimentKey}`,
+    );
+    return false;
+  }
+
+  const experimentValue = await getExperimentValue(experimentKey, 0);
+
+  if (Math.random() * 100 < experimentValue) {
+    console.debug(`Joining experiment ${experimentKey} for runner ${runnerName} with probability ${experimentValue}`);
```

Comment:
nits:
1. I haven't heard `joining` used in this context before
2. "with probability FOO" could be misinterpreted as "after you see this log line, there's a FOO% chance that the experiment will be enabled", instead of the correct interpretation that once the line is seen the experiment was definitely enabled.  I don't love my new wording too much either, so feel free to find something better

```suggestion
    console.debug(`Enabling experiment ${experimentKey} for runner ${runnerName}. Reached probability threshold of ${experimentValue}%`);
```